chains/btc/uploader/ipfs.go

```diff
@@ -0,0 +1,80 @@
+package uploader
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"mime/multipart"
+	"net/http"
+
+	"github.com/ChainSafe/sygma-relayer/config/relayer"
+)
+
+type Uploader interface {
+	Upload(proposals []map[string]interface{}) (string, error)
+}
+
+type IPFSUploader struct {
+	config relayer.UploaderConfig
+}
+
+func NewIPFSUploader(config relayer.UploaderConfig) *IPFSUploader {
+	return &IPFSUploader{config: config}
+}
+
+type IPFSResponse struct {
+	IpfsHash string `json:"IpfsHash"`
+}
+
+func (s *IPFSUploader) Upload(dataToUpload []map[string]interface{}) (string, error) {
+	// Convert proposals to JSON
+	jsonData, err := json.Marshal(dataToUpload)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a multipart form file
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", "metadata.json")
+	if err != nil {
+		return "", err
+	}
+	_, err = part.Write(jsonData)
+	if err != nil {
+		return "", err
+	}
+	writer.Close()
+
+	// Create a new request
+	req, err := http.NewRequest("POST", s.config.URL, body)
+	if err != nil {
+		return "", err
+	}
+
+	// Set the headers
+	req.Header.Add("Authorization", "Bearer "+s.config.Token)
+	req.Header.Add("Content-Type", writer.FormDataContentType())
+
+	// Make the request
+	client := &http.Client{}
+	resp, err := client.Do(req)
```

Comment:
My only question here is whether we need to add some more robustness to the HTTP client, like retrying at least a few times before giving up on uploading metadata. wdyt @mpetrun5 @tcar121293?